packages/openai-compatible/src/openai-compatible-chat-language-model.ts

```diff
@@ -0,0 +1,544 @@
+
+  /**
+Default object generation mode that should be used with this model when
+no mode is specified. Should be the mode with the best results for this
+model. `undefined` can be specified if object generation is not supported.
+  */
+  defaultObjectGenerationMode?: 'json' | 'tool' | undefined;
+};
+
+export class OpenAICompatibleChatLanguageModel implements LanguageModelV1 {
+  readonly specificationVersion = 'v1';
+
+  readonly supportsStructuredOutputs = false;
+
+  readonly modelId: OpenAICompatibleChatModelId;
+  readonly settings: OpenAICompatibleChatSettings;
+
+  private readonly config: OpenAICompatibleChatConfig;
+
+  constructor(
+    modelId: OpenAICompatibleChatModelId,
+    settings: OpenAICompatibleChatSettings,
+    config: OpenAICompatibleChatConfig,
+  ) {
+    this.modelId = modelId;
+    this.settings = settings;
+    this.config = config;
+  }
+
+  get defaultObjectGenerationMode(): 'json' | 'tool' | undefined {
+    return this.config.defaultObjectGenerationMode;
+  }
+
+  get provider(): string {
+    return this.config.provider;
+  }
+
+  private getArgs({
+    mode,
+    prompt,
+    maxTokens,
+    temperature,
+    topP,
+    topK,
+    frequencyPenalty,
+    presencePenalty,
+    stopSequences,
+    responseFormat,
+    seed,
+    stream,
+  }: Parameters<LanguageModelV1['doGenerate']>[0] & {
+    stream: boolean;
+  }) {
+    const type = mode.type;
+
+    const warnings: LanguageModelV1CallWarning[] = [];
+
+    if (topK != null) {
+      warnings.push({
+        type: 'unsupported-setting',
+        setting: 'topK',
+      });
+    }
+
+    if (
+      responseFormat != null &&
+      responseFormat.type === 'json' &&
+      responseFormat.schema != null
+    ) {
+      warnings.push({
+        type: 'unsupported-setting',
+        setting: 'responseFormat',
+        details: 'JSON response format schema is not supported',
+      });
+    }
+
+    const baseArgs = {
+      // model id:
+      model: this.modelId,
+
+      // model specific settings:
+      user: this.settings.user,
+
+      // standardized settings:
+      max_tokens: maxTokens,
+      temperature,
+      top_p: topP,
+      frequency_penalty: frequencyPenalty,
+      presence_penalty: presencePenalty,
+      stop: stopSequences,
+      seed,
+
+      // response format:
+      response_format:
+        responseFormat?.type === 'json' ? { type: 'json_object' } : undefined,
+
+      // messages:
+      messages: convertToOpenAICompatibleChatMessages(prompt),
+    };
+
+    switch (type) {
+      case 'regular': {
+        const { tools, tool_choice, toolWarnings } = prepareTools({ mode });
+        return {
+          args: {
+            ...baseArgs,
+            tools,
+            tool_choice,
+          },
+          warnings: [...warnings, ...toolWarnings],
+        };
+      }
+
+      case 'object-json': {
+        return {
+          args: {
+            ...baseArgs,
+            response_format: { type: 'json_object' },
+          },
```

Comment:
this can be much more complex with `json_schema` - it would be useful to have `structuredOutputs` setting (default to false) on `options` level

Reply:
can be separate pr

Reply:
Noted to look into subsequently.